### PR TITLE
Plans step redesign: Setup experiment and display header

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

This PR sets up the experiment for changing the signup plans step design. Users in the treatment group will see the new header title for the plans step.

**Important links**
1. [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813)
2. pau2Xa-2yE-p2 for the project thread.

**Header title if a custom domain is selected**:

Screenshot 2021-02-18 at 9 03 16 PM

**Header title if a free subdomain is selected**:

Screenshot 2021-02-18 at 9 03 37 PM

#### Testing instructions

* Open http://calypso.localhost:3000/ and assign yourself to `treatment` of the `signup_plans_step_redesign_v1` experiment in the Abacus tool.
* Head to /start and go through the signup flow. At the plans step of signup, verify that you see the new title as shown in the screenshots above. Verify for both free subdomain and custom domain selections.

### CODE REVIEW

File: client/signup/steps/plans/index.jsx [hunk 1]
```diff
@@ -202,17 +202,40 @@ export class PlansStep extends Component {
...
+	}
+
 	getSubHeaderText() {
-		const { hideFreePlan, subHeaderText, translate } = this.props;
+		const { hideFreePlan, subHeaderText, shouldShowPlansRedesign, translate } = this.props;
 
 		if ( ! hideFreePlan ) {
+			if ( shouldShowPlansRedesign ) {
+				return translate(
+					'Cancel for a full refund within 30 days risk-free. Or {{link}}start with a free site{{/link}}.',
```
Comment (dzver) [inline client/signup/steps/plans/index.jsx]: I'm not sure the number 30 is correct. According to p7DVsv-aiN-p2, it's now 14 days.

File: client/signup/steps/plans/index.jsx [hunk 2]
```diff
@@ -202,17 +202,40 @@ export class PlansStep extends Component {
...
+
 			return translate( 'Choose a plan or {{link}}start with a free site{{/link}}.', {
 				components: {
 					link: <Button onClick={ this.handleFreePlanButtonClick } borderless={ true } />,
 				},
 			} );
 		}
 
+		if ( shouldShowPlansRedesign ) {
+			return "Cancel for a full refund within 30 days. There's no risk.";
```
Comment (dzver) [inline client/signup/steps/plans/index.jsx]: 30 again.

**Review by dzver — COMMENTED** [review]
Apart from the 30-days refunds, it works as expected and looks good.

Feel free to merge after correcting the refund period (or finding that 30 is fine).

EDIT: The loading state for treatment is control, so the header blinks if I reload the page. Not sure if that's intended.

File: client/signup/main.jsx [hunk 3]
```diff
@@ -745,6 +746,7 @@ class Signup extends React.Component {
 					/>
 				) }
 				<div className="signup__steps">{ this.renderCurrentStep() }</div>
+				<QueryExperiments />
```
Comment (jessie-ross) [inline client/signup/main.jsx]: You should rather be using `<Experiment name="experimen_name" />`.

See this guide for more info: https://github.com/Automattic/experimentation-platform/wiki/Migration-guide-from-%60calypso-abtest%60#accessing-explat-state-directly

File: client/signup/steps/plans/index.jsx [hunk 4]
```diff
@@ -323,6 +348,9 @@ export default connect(
 		isLoadingExperiment: isLoading( state ),
```
Comment (jessie-ross) [inline client/signup/steps/plans/index.jsx:372]: Is this pre-existing experiment loading state covering your usage?

**Comment by jessie-ross** [comment]
Just a few comments left on the ExPlat side of things, other than that this is looking good :+1:

File: client/signup/steps/plans/index.jsx (hunk 1 above)
Comment (niranjan-uma-shankar) [reply client/signup/steps/plans/index.jsx]: Thanks for catching that. Per feedback received in pbxNRc-Hv-p2#comment-1682, I'm replacing this line with a different copy that does not mention refunds.

File: client/signup/steps/plans/index.jsx (hunk 2 above)
Comment (niranjan-uma-shankar) [reply client/signup/steps/plans/index.jsx]: Thanks for catching that. Per feedback received in pbxNRc-Hv-p2#comment-1682, I'm replacing this line with a different copy that does not mention refunds.

File: client/signup/main.jsx (hunk 3 above)
Comment (niranjan-uma-shankar) [reply client/signup/main.jsx]: I'm using the `<Experiment />` component in client/signup/steps/plans/index.jsx, so this is redundant. I've now removed this line.

File: client/signup/steps/plans/index.jsx (hunk 4 above)
Comment (niranjan-uma-shankar) [reply client/signup/steps/plans/index.jsx:372]: Yes, this covers my usage.

**Comment by niranjan-uma-shankar** [comment]
Thanks for the review.

> Feel free to merge after correcting the refund period (or finding that 30 is fine).

I've changed the sub-header to remove the refund policy as per the feedback received in pbxNRc-Hv-p2.

> The loading state for treatment is control, so the header blinks if I reload the page. Not sure if that's intended.

I've pushed in a fix for this.

**Comment by niranjan-uma-shankar** [comment]
Update: I've also implemented a change that prevents users from seeing the treatment in tablet and mobile resolutions since this experiment will be run as desktop-only.

**Review by dzver — APPROVED** [review]
LGTM

File: client/lib/plans/features-list.js [hunk 5]
```diff
@@ -253,6 +259,12 @@ export const FEATURES_LIST = {
 			),
 	},
 
+	[ constants.FEATURE_ADVANCED_SEO_EXPANDED_ABBR ]: {
+		getSlug: () => constants.FEATURE_ADVANCED_SEO_EXPANDED_ABBR,
+		getTitle: () => i18n.translate( 'Advanced SEO (Search Engine Optimisation) tools' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js:264]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Advanced SEO (Search Engine Optimization) tools' )` *ES Score: 6*

File: client/lib/plans/features-list.js [hunk 6]
```diff
@@ -542,6 +567,14 @@ export const FEATURES_LIST = {
 			),
 	},
 
+	[ constants.FEATURE_INSTALL_PLUGINS ]: {
+		getSlug: () => constants.FEATURE_INSTALL_PLUGINS,
+		getTitle: () =>
+			i18n.translate(
+				'Access to more than 50,000 WordPress plugins to extend functionality for your site'
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js:574]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Install plugins and extend functionality for your site with access to more than 50,000 WordPress plugins' )` *ES Score: 8*

File: client/my-sites/plan-features-comparison/header.jsx [hunk 7]
```diff
@@ -0,0 +1,422 @@
...
+		if ( isInSignup && isMonthlyPlan && annualPricePerMonth < rawPrice ) {
+			const discountRate = Math.round( ( 100 * ( rawPrice - annualPricePerMonth ) ) / rawPrice );
+			return translate( `Save %(discountRate)s%% by paying annually`, { args: { discountRate } } );
+		}
+
+		if ( isInSignup && ! isMonthlyPlan ) {
+			const annualPriceObj = getCurrencyObject( rawPriceAnnual, currencyCode );
+			const annualPriceText = `${ annualPriceObj.symbol }${ annualPriceObj.integer }`;
+
+			return translate( 'billed as %(price)s annually', {
```
Comment (a8ci18n) [inline client/my-sites/plan-features-comparison/header.jsx:124]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'billed as %s annually' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/new-plans-layout-signup)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5543221

Thank you @niranjan-uma-shankar for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5543221).

**Pull request merged**